apprunner-gradle-plugin/src/main/java/org/projectnessie/quarkus/gradle/QuarkusAppPlugin.java

```diff
@@ -41,7 +42,13 @@ public void apply(Project project) {
         project
             .getGradle()
             .getSharedServices()
-            .registerIfAbsent("nessie-quarkus-runner", NessieRunnerService.class, spec -> {});
+            .registerIfAbsent(
+                // Make the build-service unique per project to prevent Gradle class-cast
+                // exceptions when the plugin's reloaded within the same build using different
```

Comment:
wondering: could it be related to https://docs.gradle.org/current/userguide/build_services.html#concurrent_access_to_the_service ?

>  Setting this to 1 means that the service will be used by a single task at a time.

Reply:
I doubt, those are separte builds :(